elide-datastore/elide-datastore-jpa/src/main/java/com/yahoo/elide/datastores/jpa/JpaDataStore.java

```diff
@@ -73,15 +78,15 @@ public void populateEntityDictionary(EntityDictionary dictionary) {
     @Override
     public DataStoreTransaction beginReadTransaction() {
         EntityManager entityManager = entityManagerSupplier.get();
-        JpaTransaction transaction = readTransactionSupplier.get(entityManager);
+        JpaTransaction transaction = readTransactionSupplier.get(entityManager, jpaTransactionCancel);
         transaction.begin();
         return transaction;
     }
 
     @Override
     public DataStoreTransaction beginTransaction() {
         EntityManager entityManager = entityManagerSupplier.get();
-        JpaTransaction transaction = writeTransactionSupplier.get(entityManager);
+        JpaTransaction transaction = writeTransactionSupplier.get(entityManager, jpaTransactionCancel);
```

Comment:
You shouldn't need to pass transactionCancel as an argument to the supplier.  When the supplier is created, it can have a reference to transactionSupplier through closure.  Remove this extra argument.